src/model/model.hpp

```diff
@@ -144,6 +146,13 @@ class Model {
     /// Returns the number of constraints in the model
     [[nodiscard]] size_t NumConstraints() const { return this->constraints_.size(); }
 
+    // Returns a State object initialized from the model nodes
+    [[nodiscard]] State CreateState() const {
```

Comment:
Just making sure -- is this returning a copy of the created `State` object or moving the object itself?

Reply:
It should be moving it. Is there an annotation needed for that to happen?